scripts/aria_rdf_to_constants.xsl

```diff
@@ -0,0 +1,143 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+// Copyright 2014 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<!--
+	This can be used to generate 'axs.constants.ARIA_ROLES' when run against the ARIA taxonomy RDF at
+	    http://www.w3.org/WAI/ARIA/schemata/aria-1.rdf.
+	
+	Note that the hard-coded version has properties not found in the taxonomy:
+	    - childpresentational
+	    - namerequired
+	    - abstract
+	If necessary these can be added to the generated object in JS or hard coded in this XSLT.
+	
+	NOTE: At time of writing this is not hooked up to any part of the automated build - it is currently a standalone utility.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+	xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
+	xmlns:role="http://www.w3.org/1999/xhtml/vocab#"
+	xmlns:states="http://www.w3.org/2005/07/aaa#"
+	xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
+	xmlns:dc="http://purl.org/dc/elements/1.1/#"
+	xmlns:owl="http://www.w3.org/2002/07/owl#"
+	xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+	version="1.0">
+	
+	<xsl:output method="text"/>
+	<xsl:param name="indent" select="'    '"/>
+	<xsl:param name="newline" select="'&#xa;'"/>
+	
+	<xsl:template match="/rdf:RDF">
+		<xsl:text>/** @type {Object.&lt;string, Object&gt;} */</xsl:text>
+		<xsl:value-of select="concat($newline, 'axs.constants.ARIA_ROLES = {', $newline)"/>
+		<xsl:apply-templates select="owl:Class"/>
+		<xsl:value-of select="concat($newline, '}')"/>
+	</xsl:template>
+	
+	<xsl:template match="owl:Class">
+		<xsl:value-of select="concat($indent, '&quot;', @rdf:ID, '&quot;: {')"/>
+		<xsl:call-template name="toJsonArray">
+			<xsl:with-param name="prop">namefrom</xsl:with-param>
```

Comment:
The easiest way would be to ditch the custom transform totally (delete `aria_rdf_to_constants.xsl`) and use a generic XML to JSON transform instead - if I had been approaching this without any pre-existing code that is how I would have done it.

That would change the structure of `axs.constants.ARIA_ROLES` significantly but if you think it's worth it then I could do it.

I have used this in the past: http://www.bramstein.com/projects/xsltjson/

When considering this we should probably also ask ourselves if there should be one consistent approach to consuming these sorts of XML resources (which could also affect `parse_aria_schemas.py`) . 

Yet another option is to consume these XML resources at runtime instead of build-time and interact with the XML DOM directly from the javascript. This is what I did in ARIA Toolkit.